apps/parsley/src/context/SectionsFeatureDiscoveryContext/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { createContext, useContext, useEffect, useMemo, useState } from "react";
+import Cookies from "js-cookie";
+import { useSectionsFeatureDiscoveryAnalytics } from "analytics/sectionsFeatureDiscovery/useSectionsFeatureDiscoveryAnalytics";
+import {
+  SEEN_SECTIONS_BETA_FEATURE_MODAL,
+  SEEN_SECTIONS_BETA_GUIDE_CUE,
+} from "constants/cookies";
+
+const seenSectionsBetaGuide =
+  Cookies.get(SEEN_SECTIONS_BETA_GUIDE_CUE) === "true";
+const seenSectionsBetaFeatureModal =
+  Cookies.get(SEEN_SECTIONS_BETA_FEATURE_MODAL) === "true";
+type SectionsFeatureDiscoveryContextState = {
+  closeFeatureModal: () => void;
+  isOpenFeatureModal: boolean;
+  isOpenFirstGuideCue: boolean;
+  setIsOpenFirstGuideCue: React.Dispatch<React.SetStateAction<boolean>>;
+  isOpenSecondGuideCue: boolean;
+  setIsOpenSecondGuideCue: React.Dispatch<React.SetStateAction<boolean>>;
+  showGuideCue: boolean;
+  closeFirstGuideCue: () => void;
+  closeSecondGuideCue: () => void;
+};
+
+const SectionsFeatureDiscoveryContext =
+  createContext<SectionsFeatureDiscoveryContextState | null>(null);
+
+const useSectionsFeatureDiscoveryContext = () => {
```

Comment:
Should this be gated on the feature flag?

Reply:
I think it's more straightforward to not render the section components at all than override their default state from this hook.